### ISSUE TITLE
Data corruption when transferring small files from BlueSCSI to a Mac IIfx

### ISSUE THREAD

**Issue by smallsco** [erichelgeson/BlueSCSI-Toolbox#10]
I've been having some trouble getting the `BlueSCSI SD Transfer` app to work correctly. Small files that I attempt to transfer from my BlueSCSI to my Mac IIfx claim to be transferred successfully, but end up experiencing data corruption.

As an example - I created a plain text file on a modern Mac called `hello.txt` with the contents `Hello, World!`. I then placed this file in the `shared` directory on my BlueSCSI.  When I launch the BlueSCSI SD Transfer app on my IIfx, I can see the hello.txt file in the /shared/ directory display on the app, and I can click on the file and click the Download button, which indicates that the transfer was successful.

However, when I open the file, the contents have changed to:
```
In Main!
SCSI
```
(One interesting thing to note here - both copies of the file contain exactly 13 characters / bytes)

The debuglog.txt doesn't seem to indicate a problem:
```
In Main!
SCSISelect - id: 7 err: 2
SCSISelect - id: 6 err: 2
Found a BlueSCSI on ID: 1
Found 1 files on the SD Card...
hello.txt 13 **
Transfering files from cell: 0 size: 01
file_size: 13
bytes_remaining: 13
blocks: 0
ticks: 7 sec: 0
kbps: 0
Transfer Complete.
done.
```
(Also interesting... the first 13 characters of `debuglog.txt` are the same as those of `hello.txt`)

I then tried with three other files, in addition to `hello.txt`:
 - `goodbye.txt`, a plain text file which reads `Goodbye, World!" (15 bytes)
 - a PICT image (21 kb)
 - a Stuffit archive (44 kb)

The PICT and Stuffit files transfers completed successfully and I was able to open those on the IIfx without any problems. However, the `goodbye.txt` file was replaced by 15 characters of gibberish (random letters, accented characters, and symbols)

Lastly, I tried creating a text file with 50 paragraphs of filler text generated from https://lipsum.com/ , which came out to 31 kb, just in case the issue was related to file size instead of file format. And amazingly, it worked! So the problem isn't specific to _text_ files, but rather, to _small_ files. I've attached a screen capture showing what the files look like on the IIfx when compared to my MacBook. Note that `lipsum.txt` looks correct (save for line endings, which is expected), but `goodbye.txt` is corrupted and `hello.txt` contains the start of the debug log.

![bluescsi transfer corruption](https://user-images.githubusercontent.com/1906551/203676638-e83dd91b-36a9-443c-aa97-b58b9346f87a.png)

Things I've tried to fix it, without success:
 - Running it on my primary System 7.1.1 installation and a clean System 7.5.0 installation, and from a Mac-Rominator II ROM disk
 - Disabling extensions on System 7.1.1
 - Using a different hard disk driver (The 7.1.1 disk image has a SCSI Director Pro 4.0 driver, and then 7.5.0 disk image has an Apple driver from Disk Jockey)
 - Disconnecting everything from the SCSI bus except for the BlueSCSI
 - Removing all images from the SD card except for my 7.1.1 image (8 GB with four partitions)

Would love to hear your thoughts - should I try reformatting the SD card next, or do you think it's a different root cause?

**Comment by smallsco** [comment]
Forgot to post my LOG.txt as well:
```
BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VER: 1.1-20220927-XFER_B1-USB
DEBUG:0 SDFAT_FILE_TYPE:3
SdFat version: 2.2.0
Sd Format: exFAT
SPI speed: 50Mhz
SdFat Max FileName Length: 64
Initialized SD Card - let's go!
Sd MID:3 OID:SD
Sd Name:SR32G
Sd Date:12/2021
Sd Serial:307871495
 - HD00_512 SysAppsGamesDocs.hda HDD / 8589983744bytes / 8388656KiB / 8192MiB
Not an image: ._LOG.txt
Not an image: ._HD00_512 SysAppsGamesDocs.hda
ID:LUN0:
 0: 512:
 1:----:
 2:----:
 3:----:
 4:----:
 5:----:
 6:----:
Finished initialization of SCSI Devices - Entering main loop.
```

**Comment by erichelgeson** [comment]
This is fixed in the (unreleased) 1.0 - to come soon(tm)!
